pkg/bot/lark.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/larksuite/oapi-sdk-go/core"
+	"github.com/larksuite/oapi-sdk-go/core/tools"
+	"github.com/larksuite/oapi-sdk-go/event"
+	eventhttpserver "github.com/larksuite/oapi-sdk-go/event/http/native"
+
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/execute"
+	"github.com/infracloudio/botkube/pkg/log"
+	"github.com/infracloudio/botkube/pkg/utils"
+)
+
+const (
+	//Event lark event
+	Event = "event"
+	//ChatType lark chat type
+	ChatType = "chat_type"
+	//Text lark chat message
+	Text = "text_without_at_bot"
+	//OpenChatID lark chat id
+	OpenChatID = "open_chat_id"
+	//ChatID lark chat id
+	ChatID = "chat_id"
+	//OpenID lark user id
+	OpenID = "open_id"
+	//Users lark users
+	Users = "users"
+	//UserID lark user id
+	UserID = "user_id"
+	//Message eventType When sending a message to a chat group
+	Message = "message"
+	//AddBot eventType When a bot is added to a chat group
+	AddBot = "add_bot"
+	//P2pChatCreate eventType When a session is first created with the bot
+	P2pChatCreate = "p2p_chat_create"
+	//AddUserToChat eventType When a user is added to a chat group
+	AddUserToChat = "add_user_to_chat"
+)
+
+// LarkBot listens for user's message, execute commands and sends back the response
+type LarkBot struct {
+	AllowKubectl     bool
+	RestrictAccess   bool
+	ClusterName      string
+	DefaultNamespace string
+	Port             int
+	MessagePath      string
+	LarkClient       *utils.LarkClient
+}
+
+// NewLarkBot returns new Bot object
+func NewLarkBot(c *config.Config) Bot {
+	larkConf := c.Communications.Lark
+	appSettings := core.NewInternalAppSettings(core.SetAppCredentials(larkConf.AppID, larkConf.AppSecret),
+		core.SetAppEventKey(larkConf.VerificationToken, larkConf.EncryptKey))
+	conf := core.NewConfig(core.Domain(larkConf.Endpoint), appSettings, core.SetLoggerLevel(core.LoggerLevelError))
+	return &LarkBot{
+		AllowKubectl:     c.Settings.Kubectl.Enabled,
+		RestrictAccess:   c.Settings.Kubectl.RestrictAccess,
+		ClusterName:      c.Settings.ClusterName,
+		DefaultNamespace: c.Settings.Kubectl.DefaultNamespace,
+		Port:             c.Communications.Lark.Port,
+		MessagePath:      c.Communications.Lark.MessagePath,
+		LarkClient:       utils.NewLarkClient(conf),
+	}
+}
+
+// Execute commands sent by users
+func (l *LarkBot) Execute(e map[string]interface{}) {
+	event, err := AccessAndTypeCastToMap(Event, e)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToMap: %s", err.Error())
+		return
+	}
+
+	chatType, err := AccessAndTypeCastToString(ChatType, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	text, err := AccessAndTypeCastToString(Text, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	executor := execute.NewDefaultExecutor(text, l.AllowKubectl, l.RestrictAccess, l.DefaultNamespace,
+		l.ClusterName, config.LarkBot, "", true)
+	response := executor.Execute()
+
+	if chatType == "group" {
```

Comment:
Lets declare "group" as const and use that

Reply:
done~